File: app/src/main/java/com/anokmik/databinding/component/DataBindingComponentProvider.java
package com.anokmik.databinding.component;

import android.databinding.DataBindingComponent;

public class DataBindingComponentProvider implements DataBindingComponent {

    @Override
    public MainDataBindingComponent getMainDataBindingComponent() {
        return new MainDataBindingComponent();
    }

    @Override
    public TwoWayDataBindingComponent getTwoWayDataBindingComponent() {
        return new TwoWayDataBindingComponent();
    }

}

File: app/src/main/kotlin/com/anokmik/databinding/component/DataBindingComponentProvider.kt
package com.anokmik.databinding.component

import android.databinding.DataBindingComponent

class DataBindingComponentProvider : DataBindingComponent {

    override fun getMainDataBindingComponent(): MainDataBindingComponent {
        return MainDataBindingComponent()
    }

    override fun getTwoWayDataBindingComponent(): TwoWayDataBindingComponent {
        return TwoWayDataBindingComponent()
    }

}